src/OrchardCore/OrchardCore.ContentLocalization.Abstractions/OrchardHelperExtensions.cs

```diff
@@ -21,7 +22,11 @@ public async static Task<CultureInfo> GetContentCultureAsync(this IOrchardHelper
             var contentManager = orchardHelper.HttpContext.RequestServices.GetService<IContentManager>();
             var cultureAspect = await contentManager.PopulateAspectAsync(contentItem, new CultureAspect());
 
-            return cultureAspect.Culture;
+            // Workaround to ignore the default culture that has been set to CultureAspect.Culture
+            // for the content item which doesn't have a LocalizationPart attached
+            return contentItem.Has<LocalizationPart>()
```

Comment:
We could use `ILocalizationService` to ensure whether the `CultureAspect.Culture` is supported or not, this way no need to use `LocalizationPart` at all

Reply:
I don't see what you mean. I need to check.

Reply:
Good thing is that we're getting closer to something.

Reply:
The little issue is occur in the unit tests, coz the default culture will be supported, but I need to do more test to check what's going on

The reason is if I'm not wrong that `CultureInfo.CurrentUICulture` equals `CultureInfo.InstalledUICulture.Name`, Let us think while I finish some tests